Source/WebKit/UIProcess/WebProcessCache.cpp

```diff
@@ -59,8 +63,38 @@ WebProcessCache::WebProcessCache(WebProcessPool& processPool)
     platformInitialize();
 }
 
-bool WebProcessCache::canCacheProcess(WebProcessProxy& process) const
+bool WebProcessCache::canCacheProcess(WebProcessProxy& process, CheckSuspendedState checkSuspendedState) const
 {
+    WEBPROCESSCACHE_RELEASE_LOG("canCacheProcess: memory usage: %lu", process.processID(), process.memoryUsage());
+
+#if PLATFORM(IOS)
+    if (WTF::ramUnused() < 256 * MB) {
+        WEBPROCESSCACHE_RELEASE_LOG("canCacheProcess: Not caching process because there is not enough free memory, free RAM: %lu", process.processID(), WTF::ramUnused());
+        return false;
+    }
+
+    if (process.hasMediaActivity()) {
+        WEBPROCESSCACHE_RELEASE_LOG("canCacheProcess: Not caching process because there is media activity.", process.processID());
+        return false;
+    }
+
+    if (checkSuspendedState == CheckSuspendedState::Yes && process.throttler().currentState() != ProcessThrottleState::Suspended) {
+        WEBPROCESSCACHE_RELEASE_LOG("canCacheProcess: Not caching process because it is not suspended.", process.processID());
+        return false;
+    }
+
+    if (!process.memoryUsage()) {
+        WEBPROCESSCACHE_RELEASE_LOG("canCacheProcess: Not caching process because the process' memory usage is unknown.", process.processID());
+        return false;
+    }
+
+    constexpr unsigned memoryLimit = 100 * MB;
+    if (process.memoryUsage() > memoryLimit) {
```

Comment:
This check doesn't make a lot of sense, see comments below.

Reply:
> This check doesn't make a lot of sense, see comments below.

Thanks for reviewing, Chris! I will address these issues in an upcoming patch :)